packages/mobile/src/screens/edit-track-screen/components/CollectibleGatedAvailability.tsx

```diff
@@ -196,12 +203,14 @@ export const CollectibleGatedAvailability = ({
         true
       )
     }
-  }, [
-    premiumConditions,
-    selected,
-    setTrackAvailabilityFields,
-    selectedNFTCollection
-  ])
+  }, [selected, selectedNFTCollection, setTrackAvailabilityFields])
+
+  // Update nft collection gate when nft collection selection changes
+  useEffect(() => {
```

Comment:
the other effect is to always have the already selected nft collection e.g. when editing a track.

this effect is to only update the selected nft gate if we do have one e.g. when switching to a different nft collection or switching back to nft gate from another availability option, otherwise we don't want to set it to null/undefined; we still want to keep the previous selection.

Reply:
looking at the code again to double-check if what im saying is actually true...

Reply:
OHH I'm just now seeing that the other one uses `initialPremiumConditions` not `premiumConditions`

Reply:
I think even still, we might be able to get by on just this effect assuming the initial one is sent in premiumConditions as well?

Reply:
hmm looks like you're correct. removing the redundant effect thx